commands/deploy.go

```diff
@@ -212,7 +213,10 @@ Error: %s`, fprocessErr.Error())
 				Requests: function.Requests,
 			}
 
-			proxy.DeployFunction(function.FProcess, services.Provider.GatewayURL, function.Name, function.Image, function.RegistryAuth, function.Language, deployFlags.replace, allEnvironment, services.Provider.Network, functionConstraints, deployFlags.update, deployFlags.secrets, allLabels, functionResourceRequest1)
+			statusCode := proxy.DeployFunction(function.FProcess, services.Provider.GatewayURL, function.Name, function.Image, function.RegistryAuth, function.Language, deployFlags.replace, allEnvironment, services.Provider.Network, functionConstraints, deployFlags.update, deployFlags.secrets, allLabels, functionResourceRequest1)
+			if statusCode >= http.StatusBadRequest && statusCode <= http.StatusNetworkAuthenticationRequired {
+				os.Exit(1)
```

Comment:
This may be problematic if there is more than one function in the stack.

Reply:
Did you ever see `http.StatusNetworkAuthenticationRequired` returned from the server?

Reply:
Collect the statuses and then if there is a bad status in the list, after deploying return the error. You should see an `RunE()` function - if you return an error in the RunE function, I'm sure it will return an non-zero exit code.